Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Controller.mo

```diff
@@ -0,0 +1,2275 @@
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yHeaPreConValSta[nChi]
+    "Chiller head pressure control status setpoint"
+    annotation (Placement(transformation(extent={{800,260},{840,300}}),
+      iconTransformation(extent={{100,110},{140,150}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yHeaPreConVal[nChi](
+    final min=fill(0, nChi),
+    final max=fill(1, nChi),
+    final unit=fill("1", nChi))
+    "Head pressure control valve position"
+    annotation (Placement(transformation(extent={{800,220},{840,260}}),
+      iconTransformation(extent={{100,70},{140,110}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yConWatPumSpe[nConWatPum](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Condenser water pump speed setpoint"
+    annotation (Placement(transformation(extent={{800,130},{840,170}}),
+        iconTransformation(extent={{100,40},{140,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatMinFloSet(
+    final quantity="VolumeFlowRate",
+    final unit="m3/s",
+    final min=0) "Chilled water minimum flow setpoint"
+    annotation (Placement(transformation(extent={{800,100},{840,140}}),
+      iconTransformation(extent={{100,12},{140,52}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yConWatPum[nConWatPum]
+    "Status setpoint of condenser water pump"
+    annotation (Placement(transformation(extent={{800,70},{840,110}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yChiWatIsoVal[nChi](
+    final unit=fill("1", nChi),
+    final min=fill(0, nChi),
+    final max=fill(1, nChi))
+    "Chiller isolation valve position setpoints"
+    annotation (Placement(transformation(extent={{800,-20},{840,20}}),
+        iconTransformation(extent={{100,-90},{140,-50}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yReaChiDemLim
+    if need_reduceChillerDemand
+    "Release chiller demand limit, normally true"
+    annotation (Placement(transformation(extent={{800,-230},{840,-190}}),
+        iconTransformation(extent={{100,-60},{140,-20}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yMinValPosSet(
+    final min=0,
+    final max=1,
+    final unit="1") "Chilled water minimum flow bypass valve position setpoint"
+    annotation (Placement(transformation(extent={{800,-340},{840,-300}}),
+        iconTransformation(extent={{100,-120},{140,-80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerOutput yNumCel
+    "Total number of enabled cooling tower cells"
+    annotation(Placement(transformation(extent={{800,-560},{840,-520}}),
+      iconTransformation(extent={{100,-170},{140,-130}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yTowCelIsoVal[nTowCel](
+    final min=fill(0, nTowCel),
+    final max=fill(1, nTowCel),
+    final unit=fill("1", nTowCel))
+    "Cooling tower cells isolation valve position"
```

Comment:
This should be
  - Boolean, and
  - conditional to a configuration parameter (as CT may not have isolation valves).